storm-webapp/src/main/java/org/apache/storm/daemon/logviewer/LogviewerServer.java

```diff
@@ -0,0 +1,182 @@
+            final String httpsKeyPassword = (String) (conf.get(DaemonConfig.LOGVIEWER_HTTPS_KEY_PASSWORD));
+            final String httpsTsPath = (String) (conf.get(DaemonConfig.LOGVIEWER_HTTPS_TRUSTSTORE_PATH));
+            final String httpsTsPassword = (String) (conf.get(DaemonConfig.LOGVIEWER_HTTPS_TRUSTSTORE_PASSWORD));
+            final String httpsTsType = (String) (conf.get(DaemonConfig.LOGVIEWER_HTTPS_TRUSTSTORE_TYPE));
+            final Boolean httpsWantClientAuth = (Boolean) (conf.get(DaemonConfig.LOGVIEWER_HTTPS_WANT_CLIENT_AUTH));
+            final Boolean httpsNeedClientAuth = (Boolean) (conf.get(DaemonConfig.LOGVIEWER_HTTPS_NEED_CLIENT_AUTH));
+
+            //TODO a better way to do this would be great.
+            LogviewerApplication.setup(conf);
+            ret = UIHelpers.jettyCreateServer(logviewerHttpPort, null, httpsPort);
+
+            UIHelpers.configSsl(ret, httpsPort, httpsKsPath, httpsKsPassword, httpsKsType, httpsKeyPassword,
+                    httpsTsPath, httpsTsPassword, httpsTsType, httpsNeedClientAuth, httpsWantClientAuth);
+
+            ServletContextHandler context = new ServletContextHandler(ServletContextHandler.NO_SESSIONS);
+            try {
+                context.setBaseResource(Resource.newResource(STATIC_RESOURCE_DIRECTORY_PATH));
+            } catch (IOException e) {
+                throw new RuntimeException("Can't locate static resource directory " + STATIC_RESOURCE_DIRECTORY_PATH);
+            }
+
+            context.setContextPath("/");
+            ret.setHandler(context);
+
+            ServletHolder holderPwd = new ServletHolder("default", DefaultServlet.class);
+            holderPwd.setInitOrder(1);
+            context.addServlet(holderPwd,"/");
+
+            ServletHolder jerseyServlet = context.addServlet(ServletContainer.class, "/api/v1/*");
+            jerseyServlet.setInitOrder(2);
+            jerseyServlet.setInitParameter("javax.ws.rs.Application", LogviewerApplication.class.getName());
+
+            UIHelpers.configFilters(context, filterConfigurations);
+        }
+        return ret;
+    }
+
+    private final Server httpServer;
+    private boolean closed = false;
+
+    /**
+     * Constructor.
+     * @param conf Logviewer conf for the servers
+     */
+    public LogviewerServer(Map<String, Object> conf) {
+        httpServer = mkHttpServer(conf);
+    }
+
+    @VisibleForTesting
+    void start() throws Exception {
+        LOG.info("Starting Logviewer...");
+        if (httpServer != null) {
+            httpServer.start();
+        }
+    }
+
+    @VisibleForTesting
+    void awaitTermination() throws InterruptedException {
+        httpServer.join();
+    }
+
+    @Override
+    public synchronized void close() {
+        if (!closed) {
+            //This is kind of useless...
+            meterShutdownCalls.mark();
+
+            //TODO this is causing issues...
```

Comment:
I thought it looked familiar.  I'll file a follow on JIRA to try and clean it up once we have everything ported over to java.